util/dataset.go

```diff
@@ -24,6 +24,13 @@ const (
 	DirIngestPipeline = "ingest-pipeline"
 )
 
+var ValidTypes = map[string]interface{}{
```

Comment:
I wonder if `ValidTypes` can be package visible only?

Reply:
yes, changed.